ami/ml/orchestration/nats_queue.py

```
@@ -330,9 +338,134 @@ async def delete_stream(self, job_id: int) -> bool:
+                        # Look up the actual message by sequence to get task ID
+                        try:
+                            job_msg = await self.js.get_msg(stream_name, stream_seq)
+
+                            if job_msg and job_msg.data:
+                                task_data = json.loads(job_msg.data.decode())
+
+                                if "image_id" in task_data:
+                                    dead_letter_ids.append(str(task_data["image_id"]))
+                                else:
+                                    logger.warning(f"No image_id found in task data: {task_data}")
+                        except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)